rand_distr/src/normal_inverse_gaussian.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{Distribution, Float, InverseGaussian, Standard, StandardNormal};
+use rand::Rng;
+
+/// Error type returned from `NormalInverseGaussian::new`
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// `alpha <= 0` or `nan`.
+    AlphaNegativeOrNull,
+    /// `|beta| >= alpha` or `nan`.
+    AbsoluteBetaLessThanAlpha,
+}
+
+/// The [normal-inverse Gaussian distribution](https://en.wikipedia.org/wiki/Normal-inverse_Gaussian_distribution)
+#[derive(Debug)]
+pub struct NormalInverseGaussian<N> {
+    alpha: N,
+    beta: N,
+    inverse_gaussian: InverseGaussian<N>,
+}
+
+impl<N: Float> NormalInverseGaussian<N>
+where StandardNormal: Distribution<N>
+{
+    /// Construct a new `NormalInverseGaussian` distribution with the given alpha (tail heaviness) and
+    /// beta (asymmetry) parameters.
+    pub fn new(alpha: N, beta: N) -> Result<NormalInverseGaussian<N>, Error> {
```

Comment:
I guess that this is equivalent to the definition on WP when μ=0, δ=1?

But then the mean of the samples doesn't match (`β/γ` vs `β/γ + 1/(2√γ)`).

Reply:
Yes that's right. I was wondering if I needed to add the `location` / `scale` parameters in the distribution definition or if we let the users do it themselves.
About the mean, I think it does match because you multiply by the standard normal distribution which has a zero mean.

Reply:
You're right.